Refacto & some new things

pmachart opened the pull request:
First, thanks a lot for sharing this ! :+1: 
It inspired me to use `rofi` some more :)

General refacto :
- wrap in functions to use local variables
- integrate confirmation into main script
- use recommended safe syntax for variables (shellcheck)
- #/usr/bin/env bash to make sure the correct bash in used
- escaped common binaries (such as `cat`) to ensure unaliased usage
(eg: if `cat` has an alias on top of it, using `\cat` will use the basic `cat` instead. Many people alias their `cat` for colors or other things.)

New things :
- new option `unsafe` : makes `yes` focused by default
- new option `message` : you can supply a custom string instead of the default "Confirm ?"
- supplied a sample tasks.json file with various examples

pmachart committed:
refacto & new message field

moustacheful commented:
Hey @pmachart this looks great! I don't know a lot about bash, so I appreciate this. I'll take a look and let you know if there's any comments, otherwise I'll merge this right in.

Thanks!

moustacheful commented on myrmidon.sh:
```diff
@@ -1,28 +1,52 @@
+  OPTIONS="No\\nYes";
+
+  UNSAFE=$(echo "${TASK}" | jq ".unsafe")
+  [[ ${UNSAFE} == "true" ]] && OPTIONS="Yes\\nNo";
```
Is this to provide inverted options, right? default to Yes instead of No?
If so, the naming seems a bit backwards to me. like this should be the safe, unsafe should be the original behavior.

Maybe this could be set as a different name associated with confirm, maybe
```
confirm: true
confirmInvert: true
```

I'm not sure I like it though!

moustacheful replied:
Another option could be setting confirm to be either true or an object with options, for instance:

```
confirm: {
   message: "This is dangerous! are you sure",
   invert: true
}
```
This makes it more explicit what those properties do, which are directly related to the `confirm` mechanism.

pmachart replied:
the idea behind the `unsafe` keyword actually comes from... my camera's settings :)
by default when deleting a picture, there's a yes/no confirmation with the focus on the `no`
there's a setting in the menu that they called 'unsafe deleting' that makes the 'yes' selected by default instead.
so yeah. i guess `confirmInvert` makes more sense to most people.
a whole confirm object is also an interesting idea !

moustacheful commented on tasks.json:
```diff
@@ -0,0 +1,30 @@
+[
```
I'd remove this whole file, an example with the additional properties in the readme file should be enough. Since we can't really assume what's on the users' system.

pmachart replied:
i just wanted to show what the config file would look like with the options i added.
guess i could have just edited the readme.

moustacheful reviewed (commented):
Hey @pmachart thanks again for contributing!
I made some comments regarding the new properties that you added, they look quite helpful, but I think naming is a bit off.

Let me know what you think!

pmachart commented:
> Hey @pmachart this looks great! I don't know a lot about bash, so I appreciate this. I'll take a look and let you know if there's any comments, otherwise I'll merge this right in.

Hey !
Glad you like it. Thanks for taking the time to look at this !
I answered your comments. Don't hesitate to ask anything !

edit: i'm making a few changes at the moment, will update the PR soon !
edit2: pushed a second commit. will update with a few comments about it asap :)
edit3: here goes :

- added options to configure `yes` and `no`
- renamed `unsafe` to `invert`
- refactored with `confirm` as an optional object for advanced configuration of the confirmation prompt
- confirmation config stays default when confirm is set to `true` in order to maintain retro-compatibility with people's older config files
- deleted sample json
- updated readme with sample json and a couple changes

- used the `--exit-status` super useful option from `jq` to make defaulting the confirmation message more straightforward

- simplified checking if confirmation is needed by simply exiting if the confirmation function doesn't exit with 0 (instead of the previous if/else with the repeated eval statement)

pmachart committed:
more refacto & more new things !
